src/features/Search/useQueryParameter.js

```diff
@@ -0,0 +1,11 @@
+import { useLocation } from "react-router-dom";
+
+const useQueryParameter = (paramName) => {
+  const location = useLocation();
+  const searchParams = new URLSearchParams(location.search);
+  const query = searchParams.get(paramName);
+
+  return query;
+};
+
+export default useQueryParameter;
```

Comment:
Wszystko tutaj dziala jak nalezy, podrzucam krotszy zapis, mozesz z niego skorzystac lub nie poprostu chcialem pokazac, ze da sie to inaczej zapisac :) : 

const useQueryParameter = (paramName) => {
  const location = useLocation();
  return new URLSearchParams(location.search).get(paramName);
};

export default useQueryParameter;